pkg/aggregator/host_tag_provider.go

```diff
@@ -0,0 +1,52 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//nolint:revive
+package aggregator
+
+import (
+	"context"
+	"sync"
+
+	hostMetadataUtils "github.com/DataDog/datadog-agent/comp/metadata/host/hostimpl/hosttags"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+
+	"github.com/benbjohnson/clock"
+)
+
+type HostTagProvider struct {
+	hostTags []string
+	sync.RWMutex
+}
+
+func NewHostTagProvider() *HostTagProvider {
+	return newHostTagProviderWithClock(clock.New())
+}
+
+func newHostTagProviderWithClock(clock clock.Clock) *HostTagProvider {
+	p := &HostTagProvider{
+		hostTags: nil,
+	}
+
+	duration := pkgconfigsetup.Datadog().GetDuration("expected_tags_duration")
+	if pkgconfigsetup.Datadog().GetDuration("expected_tags_duration") > 0 {
```

Comment:
Let's retain the log line from `internal/host/host.go`
`log.Debugf("Adding host tags to metrics for %v", duration)`

And maybe also have a similar log line that is printed when the tags expire?